### PR TITLE
Migrate ZIOSpec to ZIO Test (#1647)

### PR DESCRIPTION
**Pull request by chlin501**
Try migrating ZIOSpec to ZIO Test. But I am not sure if this is the right way to do that, so attempting to start from a small patch first. Please let me know if any issues.

### CODE REVIEW

File: core-tests/shared/src/test/scala/zio/ZIOSpec.scala [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-  def oneFailure =
-    for {
-      f1     <- IO.fail("error1").fork
-      f2     <- IO.succeed("success1").fork
-      errors <- f1.zip(f2).join.parallelErrors[String].flip
-    } yield errors must_=== ::("error1", List())
-}
+object ZIOSpec
+    extends ZIOBaseSpec(
+      suite("ZIOSpec")(
```
Comment (LGLO) [inline core-tests/shared/src/test/scala/zio/ZIOSpec.scala]: I think that whole suite is about "ZIO#parallelErrors", so I would rename it.

File: core-tests/shared/src/test/scala/zio/ZIOSpec.scala [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+        testM("oneFailure") {
+          for {
+            f1     <- IO.fail("error1").fork
+            f2     <- IO.succeed("success1").fork
+            errors <- f1.zip(f2).join.parallelErrors[String].flip
+          } yield assert(errors, equalTo(::("error1", List())))
```
Comment (LGLO) [inline core-tests/shared/src/test/scala/zio/ZIOSpec.scala]: Nitpick: first time I see prefix `::` together with `List()`, could you, please, use something more standard: `"error1" :: Nil` or `List("error1")`?
I'm aware it was like it before.

**Comment by chlin501** [comment]
Please let me know if some more places that I can improve. I am happy to do that.

**Comment by ghostdogpr** [comment]
@chlin501 thanks!

**Pull request merged**